src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/PartBRecordService.kt

```diff
@@ -78,13 +82,21 @@ class PartBRecordService(
 
   private fun scanAndStoreDocuments(request: PartBRecordRequest, recallId: RecallId, currentUserId: UserId):
     Map<DocumentCategory, Pair<FileName, Result<DocumentId, VirusScanResult>>> {
-    val res = mutableMapOf(
-      scanAndStore(PART_B_RISK_REPORT, request.details, request.partBFileContent, request.partBFileName, recallId, currentUserId),
-      scanAndStore(PART_B_EMAIL_FROM_PROBATION, request.details, request.emailFileContent, request.emailFileName, recallId, currentUserId)
-    )
-    request.oasysFileContent?.let {
-      val oasysPair = scanAndStore(OASYS_RISK_ASSESSMENT, "Uploaded alongside Part B", it, request.oasysFileName!!, recallId, currentUserId)
-      res.put(oasysPair.first, oasysPair.second)
+    val res = mutableMapOf<DocumentCategory, Pair<FileName, Result<DocumentId, VirusScanResult>>>()
+    runBlocking {
+      val asyncs = mutableListOf<Deferred<Pair<DocumentCategory, Pair<FileName, Result<DocumentId, VirusScanResult>>>>>()
+
+      asyncs.add(async { scanAndStore(PART_B_RISK_REPORT, request.details, request.partBFileContent, request.partBFileName, recallId, currentUserId) })
```

Comment:
(Retrospectively) At the end of our call-chain to gotenberg we use Spring reactive WebClient where the response object exposes e.g. `BodyToMono` to pass a reactive response back along the chain.  How "reactive" this is over Tomcat I don't know; but there are comments around about it being incomplete and moving to Netty instead: https://mojdt.slack.com/archives/C012FNNQ4A3/p1644585472288789
For clamav we use the ClamavClient which has no reactive elements.  Likewise our calls to s3 have no reactive component.  In this context I found this use of coroutines easier to apply.  The default on async() creation is to start so IMO this will parallelise these functions and the resulting calls to clamav and s3.